### PR TITLE
Refactor backend

### PR DESCRIPTION
**Pull request by falapen**

### CODE REVIEW

File: backend/news-service/src/utils.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  return nextDay;
+}
+
+function getIncludedTimespans(newFrom, newUntil, timespans) {
+  let tentFrom = null;
+  let tentUntil = null;
+  let placedFrom = false;
+  const includedTimespans = [];
+  for (let i = 0; i < timespans.length; i += 1) {
+    tentFrom = new Date(timespans[i].from);
```
Comment (antoncarlsson) [inline backend/news-service/src/utils.js]: Should be `const`

File: backend/news-service/src/utils.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    tentUntil = new Date(timespans[i].until);
```
Comment (antoncarlsson) [inline backend/news-service/src/utils.js]: Should be `const`

File: backend/news-service/src/utils.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+function getPreviousDay(date) {
+  const nextDay = new Date(date);
+  nextDay.setDate(nextDay.getDate() - 1);
+  return nextDay;
+}
+
+function getIncludedTimespans(newFrom, newUntil, timespans) {
+  let tentFrom = null;
+  let tentUntil = null;
+  let placedFrom = false;
```
Comment (antoncarlsson) [inline backend/news-service/src/utils.js]: Should be declared inside the loop.

File: backend/news-service/src/utils.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  }
+  return includedTimespans;
+}
+
+function getNewTimespan(inputFrom, inputUntil, timespans) {
+  if (timespans.length === 0) return { from: inputFrom, until: inputUntil };
+  const firstFrom = new Date(timespans[0].from);
+  const lastUntil = new Date(timespans[timespans.length - 1].until);
+
+  let newFrom = null;
```
Comment (antoncarlsson) [inline backend/news-service/src/utils.js]: Could be declared as `const` if you assign it with a ternary operator.

File: backend/news-service/src/utils.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  let newUntil = null;
```
Comment (antoncarlsson) [inline backend/news-service/src/utils.js]: Could be declared as `const` if you assign it with a ternary operator.

File: backend/news-service/src/utils.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    tentFrom = new Date(timespans[i].from);
+    tentUntil = new Date(timespans[i].until);
+    if (!placedFrom) {
+      if (newFrom > getNextDay(tentUntil)) {
+        continue;
+      } else {
+        placedFrom = true;
+      }
+    }
+    if (placedFrom) {
```
Comment (antoncarlsson) [inline backend/news-service/src/utils.js]: Will always be true.

File: backend/news-service/src/utils.js (hunk 3 above)
Comment (falapen) [reply backend/news-service/src/utils.js]: Fixed

File: backend/news-service/src/utils.js (hunk 1 above)
Comment (falapen) [reply backend/news-service/src/utils.js]: fixed

File: backend/news-service/src/utils.js [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+function getIncludedTimespans(newFrom, newUntil, timespans) {
+  let tentFrom = null;
+  let tentUntil = null;
+  let placedFrom = false;
+  const includedTimespans = [];
+  for (let i = 0; i < timespans.length; i += 1) {
+    tentFrom = new Date(timespans[i].from);
+    tentUntil = new Date(timespans[i].until);
+    if (!placedFrom) {
+      if (newFrom > getNextDay(tentUntil)) {
```
Comment (antoncarlsson) [inline backend/news-service/src/utils.js]: Could be combined with the statement above.

File: backend/news-service/src/utils.js (hunk 2 above)
Comment (falapen) [reply backend/news-service/src/utils.js]: fixed

File: backend/news-service/src/utils.js [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+        continue;
+      } else {
```
Comment (antoncarlsson) [inline backend/news-service/src/utils.js]: If the comment above is resolved this else case is unnecessary.

File: backend/news-service/src/Requests.js [hunk 9]
```diff
@@ -77,17 +61,17 @@ function updateRequestCompletion(request) {
...
 }
 
 // Add a request to the requests-object and check the request for completion.
 function addRequest(requestId, requestedResource) {
-  setInterval(() => {
-    console.log(requests);
-  }, 5000);
+  // setInterval(() => {
+  //  console.log(requests);
+  // }, 5000);
```
Comment (antoncarlsson) [inline backend/news-service/src/Requests.js]: Commented out code.

File: backend/news-service/src/utils.js (hunk 4 above)
Comment (falapen) [reply backend/news-service/src/utils.js]: No thank you

File: backend/news-service/src/utils.js (hunk 5 above)
Comment (falapen) [reply backend/news-service/src/utils.js]: No thank you

File: backend/middleware/src/requests/Requests.js [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+  const requestsToRemove = [];
+  for (const requestId in requests) {
+    if (requests.hasOwnProperty(requestId)) {
+      if (requests[requestId].clientId === clientId) {
+        requestsToRemove.push(requestId);
+      }
+    }
+  }
+  for (const requestId in requestsToRemove) {
+    delete requests[requestId];
```
Comment (antoncarlsson) [inline backend/middleware/src/requests/Requests.js]: Should use the removeRequest method.

File: backend/news-service/src/utils.js (hunk 4 above)
Comment (falapen) [reply backend/news-service/src/utils.js]: Nevermind fixed

File: backend/news-service/src/utils.js (hunk 5 above)
Comment (falapen) [reply backend/news-service/src/utils.js]: Nevermind, fixed

File: backend/news-service/src/utils.js (hunk 6 above)
Comment (falapen) [reply backend/news-service/src/utils.js]: Leave me alone

File: backend/news-service/src/utils.js (hunk 7 above)
Comment (falapen) [reply backend/news-service/src/utils.js]: Go away

File: backend/news-service/src/utils.js (hunk 8 above)
Comment (falapen) [reply backend/news-service/src/utils.js]: Not touching these things right now

**Pull request merged**